packages/core/src/compiler.ts

```diff
@@ -303,21 +309,28 @@ export default class Compiler {
     }
   }
 
-  setFormatter(key: string) {
+  setFormatter(key: string, parentKey?: string) {
     if (this.runtimeIncludes(key, 'formatter')) return;
-    let cf = this.options.customFormatters[key];
+    const cf = this.options.customFormatters[parentKey || key];
     if (cf) {
-      if (typeof cf === 'function') cf = { formatter: cf };
+      const cfo = typeof cf === 'function' ? { formatter: cf } : cf;
+
       this.runtime[key] = Object.assign(
-        cf.formatter,
+        cloneFunction(cfo.formatter),
         { type: 'formatter' } as const,
         'module' in cf && cf.module && cf.id
-          ? { id: identifier(cf.id), module: cf.module }
+          ? {
+              id: identifier(cf.id),
+              module:
+                typeof cf.module === 'function' && this.plural.locale
```

Comment:
Checking `this.plural.locale` is incorrect here. If it may be null, then either it needs to be coerced to an empty string, or the function needs to be able to deal with a null input.

Fixing this will let you drop the `compile-module.ts` change.

Reply:
ack, fixed 72396da